### PR TITLE
add allowUrls and denyUrls for Flutter

### PR DESCRIPTION
**Pull request by martinhaintz**
## DESCRIBE YOUR PR
Added allowUrls and denyUrls for Flutter, which is implemented in this PR: [2227](https://github.com/getsentry/sentry-dart/pull/2227#pullrequestreview-2234892068)

## IS YOUR CHANGE URGENT?  

Help us prioritize incoming PRs by letting us know when the change needs to go live.
- [ ] Urgent deadline (GA date, etc.): 
- [ ] Other deadline: 
- [x] None: Not urgent, can wait up to 1 week+

## PRE-MERGE CHECKLIST

*Make sure you've checked the following before merging your changes:*

- [x] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs (subject matter experts)
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

### CODE REVIEW

File: docs/platforms/flutter/configuration/filtering.mdx [hunk 1]
```diff
@@ -65,6 +65,28 @@ When a string or a non-error object is raised, Sentry creates a synthetic except
 
 <PlatformContent includePath="configuration/breadcrumb-hints" />
 
+#### Using `allowUrls` (Web only)
+
+Based on the URL, you can narrow down whether events should be sent to Sentry.
+Using `allowUrls`, events are only sent if the URL matches the pattern specified in `allowUrls`.
+`allowUrls` uses regular expression to compare.
```
Comment (vivianyentran) [inline docs/platforms/flutter/configuration/filtering.mdx:72]:
```suggestion
`allowUrls` accepts regular expressions.
```

File: docs/platforms/flutter/configuration/filtering.mdx [hunk 2]
```diff
@@ -65,6 +65,28 @@ When a string or a non-error object is raised, Sentry creates a synthetic except
...
+
+If used on a platform other than Web, this setting will be ignored.
+
+<PlatformContent includePath="configuration/allow-urls" />
+
+#### Using `denyUrls` (Web only)
+
+You can exclude events sent to Sentry based on the URL.
+Using `denyUrls`, events are ignored if the URL matches the pattern specified in `denyUrls`.
+`denyUrls` uses regular expression to compare.
```
Comment (vivianyentran) [inline docs/platforms/flutter/configuration/filtering.mdx:82]:
```suggestion
`denyUrls` accepts regular expressions.
```

File: docs/platforms/flutter/configuration/filtering.mdx [hunk 3]
```diff
@@ -65,6 +65,28 @@ When a string or a non-error object is raised, Sentry creates a synthetic except
...
+
+In combination with `allowUrls` you can block subdomains of the domains listed in `allowUrls`.
```
Comment (vivianyentran) [inline docs/platforms/flutter/configuration/filtering.mdx:84]:
```suggestion
When used in combination with `allowUrls`, you can block specific subdomains of the domains listed in `allowUrls`.
```

**Review by vivianyentran — APPROVED** [review]
LGTM, thanks for adding! 
@buenaflor  Can you help review the code samples?

File: platform-includes/configuration/allow-urls/dart.mdx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+```dart
+import 'package:sentry/sentry.dart';
+
+Future<void> main() async {
+  await Sentry.init(
+    (options) {
+      options.dsn = '___PUBLIC_DSN___';
+      options.allowUrls = ["^https://sentry.com.*\$", "my-custom-domain"];
+      ...
```
Comment (buenaflor) [inline platform-includes/configuration/allow-urls/dart.mdx:9]:
```suggestion
```

let's make the snippets compileable so users can copy paste without errors

File: platform-includes/configuration/deny-urls/dart.mdx [hunk 5]
```diff
@@ -0,0 +1,14 @@
+```dart
+import 'package:sentry/sentry.dart';
+
+Future<void> main() async {
+  await Sentry.init(
+    (options) {
+      options.dsn = '___PUBLIC_DSN___';
+      options.denyUrls = ["^.*ends-with-this\$", "denied-url"];
+      ...
```
Comment (buenaflor) [inline platform-includes/configuration/deny-urls/dart.mdx:9]:
```suggestion
```

same as above

**Review by buenaflor — APPROVED** [review]
added comments about the code snippet

**Pull request merged**